File: leetCode/26. 删除排序数组中的重复项.cpp
#include<iostream>
#include<stack>
#include<queue>
#include<vector>
using namespace std;
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int len = nums.size();
        if (len == 0) {
            return 0;
        }
        int p = 0,q = 1;
        while (q < len) {
            if (nums[p] != nums[q]) {
                nums[p + 1] = nums[q];
                p++;
            }
            q++;
        }
        return p + 1;
    }
};

File: src/leetCode/26. 删除排序数组中的重复项.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if(nums.length<1){
        return 0
    }
    let p = 0,q = 1;
    while(q<nums.length){
        if(nums[p] !== nums[q]){
            nums[p+1] = nums[q]
            p++;
        }
        q++;
    }
    return p+1;
};